lib/cancel/CancelToken.js

```diff
@@ -35,6 +35,9 @@ function CancelToken(executor) {
  */
 CancelToken.prototype.throwIfRequested = function throwIfRequested() {
   if (this.reason) {
+    if (typeof this.reason === typeof 'string') {
```

Comment:
:bulb: consider nuking the `typeof` on the right

```
if (typeof this.reason === 'string') 
```

see also: [this SO question](https://stackoverflow.com/questions/4059147/check-if-a-variable-is-a-string-in-javascript)